app/core/consensus/committeemanager.py

```diff
@@ -201,23 +201,30 @@ def get_committee_for_current_block(last_block=None):
     global miner_committee_cache
     if last_block is None:
         last_block = get_last_block_hash()
-
     if not last_block:
         return [{'wallet_address': SENTINEL_NODE_WALLET}]
 
     if is_miner_committee_cached(last_block['hash']):
         return miner_committee_cache['current_committee']
 
-    miners = get_eligible_miners()
-
-    if len(miners) < MINIMUM_ACCEPTANCE_VOTES:
-        logger.info("Current committee cannot form consensus. Using sentinel node.")
-        return [{'wallet_address': SENTINEL_NODE_WALLET}]
+    cutfoff_block = last_block['index'] - 1000
+    if cutfoff_block > FORK_COMMITTEE_SELECTION_SCORE_BASED:
+        # use combined stake + trust scores as weights
+        miners = get_eligible_miners_with_data()  
+        if len(miners) < MINIMUM_ACCEPTANCE_VOTES:
+            logger.info("Current committee cannot form consensus. Using sentinel node.")
+            return [{'wallet_address': SENTINEL_NODE_WALLET}]
+        weights = get_combined_scores(miners)
+    else:    
+        miners = get_eligible_miners()
+        if len(miners) < MINIMUM_ACCEPTANCE_VOTES:
```

Comment:
```
if len(miners) < MINIMUM_ACCEPTANCE_VOTES:
            logger.info("Current committee cannot form consensus. Using sentinel node.")
            return [{'wallet_address': SENTINEL_NODE_WALLET}]
```

duplicate code. can we keep this common?

Reply:
Yes min acceptance check can be kept as common later but it add extra step of get_combined_scores( wallet fetch, normalise and calculate the scores) unnecessarily if min votes accepted isn't reached. So it was a trade off.